pkg/service/contract/utils.go

```diff
@@ -0,0 +1,261 @@
+func ExtractQueryParams(rawURL string) (map[string]string, error) {
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+	queryParams := parsedURL.Query()
+	params := make(map[string]string)
+	for key, values := range queryParams {
+		if len(values) > 0 {
+			// Take the first value if multiple are present
+			params[key] = values[0]
+		}
+	}
+	return params, nil
+}
+
+// GenerateDummyNamesForIdentifiers generates dummy names for the path identifiers.
+func GenerateDummyNamesForIdentifiers(identifiers []string) map[string]string {
+	dummyNames := make(map[string]string)
+	for i, id := range identifiers {
+		dummyName := fmt.Sprintf("param%d", i+1)
+		dummyNames[dummyName] = id
+	}
+	return dummyNames
+}
+func AppendInParameters(parameters []models.Parameter, inParameters map[string]string, paramType string) []models.Parameter {
+
+	for key, value := range inParameters {
+		parameters = append(parameters, models.Parameter{
+			Name:     key,
+			In:       paramType,
+			Required: true,
+			Schema:   models.ParamSchema{Type: "string"},
+			Example:  value,
+		})
+	}
+
+	return parameters
+}
+
+// ReplacePathIdentifiers replaces numeric identifiers in the path with their corresponding dummy names.
+func ReplacePathIdentifiers(path string, dummyNames map[string]string) string {
+	segments := strings.Split(path, "/")
+	var replacedPath []string
+	for _, segment := range segments {
+		if segment != "" {
+			// Check if the segment is numeric (integer or float)
+			if isNumeric(segment) {
+				dummyName := ""
+				for key, value := range dummyNames {
+					if value == segment {
+						// i want to put '{' and '}' around the key
+						dummyName = "{" + key + "}"
+						break
+					}
+				}
+				if dummyName != "" {
+					replacedPath = append(replacedPath, dummyName)
+				} else {
+					replacedPath = append(replacedPath, segment)
+				}
+			} else {
+				replacedPath = append(replacedPath, segment)
+			}
+		}
+	}
+	finalPath := strings.Join(replacedPath, "/")
+	// Add slash at the beginning of the path
+	finalPath = "/" + finalPath
+	return finalPath
+}
+
+func generateUniqueID() string {
+	b := make([]byte, 16)
+	_, err := rand.Read(b)
+	if err != nil {
+		// handle error
+		return ""
+	}
+	return hex.EncodeToString(b) + "-" + time.Now().Format("20060102150405")
+}
+
+func checkConfigFile(servicesMapping map[string][]string) error {
+	// Check if the size of servicesMapping is less than 1
+	if len(servicesMapping) < 1 {
+		return fmt.Errorf("services mapping must contain at least 1 services")
+	}
+	return nil
+}
+
+func saveServiceMappings(servicesMapping config.Config, filePath string) error {
+	// Marshal the services mapping to YAML
+	servicesMappingYAML, err := yamlLib.Marshal(servicesMapping)
+	if err != nil {
+		return fmt.Errorf("failed to marshal services mapping: %w", err)
+	}
+
+	// Write the services mapping to the specified file path
```

Comment:
I think the path is wrong here. it should be `keploy/schema/serviceMappings.yaml`,but it looks like it will be saved in keploy folder only.
As I told you earlier also, we don't have to save the entire config file but just the self and service mappings